File: src/domain/model/character.ts
import {Entity} from './shared.model';

export interface Character extends Entity {
  name: string;
  href?: string;
  bookIds: string[];
  planet?: string;
  description?: string;
  universe: string;
  ethnicity?: string;
}

export const equalCharacters = (character1: Character, character2: Character): boolean => character1.id === character2.id;

File: src/infrastructure/python/model/character.py
class Character:
    def __init__(self, name):
        self.id = name
        self.name = name
        self.href = ""
        self.bookIds = ["mistborn1"]
        self.planet = "scadrial"
        self.description = ""

    def __str__(self):
      return self.name

    def __hash__(self):
      return hash(self.id)

    def __eq__(self, character):
        return character.id == self.id
